src/optimize.jl

```diff
@@ -0,0 +1,122 @@
+#######################################################
+# training loop for variational objectives 
+#######################################################
+function pm_next!(pm, stats::NamedTuple)
+    return ProgressMeter.next!(pm; showvalues=[tuple(s...) for s in pairs(stats)])
```

Comment:
To avoid the allocations:
```suggestion
    return ProgressMeter.next!(pm; showvalues=map(tuple, keys(stats), values(stats)))
```